### PR TITLE
Upgrade to Python 3.8

### PR DESCRIPTION
**Pull request by jon-betts**
For: https://github.com/hypothesis/h/issues/6600

This shouldn't really work until we sort out the upgrade path for `ws4py`: https://github.com/hypothesis/h-ws4py/pull/2, which might involve forking it for ourselves as the project isn't very active.

### CODE REVIEW

**Comment by jon-betts** [comment]
Some more fun in the docker image:

```
+ apk add --no-cache build-base libffi-dev postgresql-dev python-dev
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/community/x86_64/APKINDEX.tar.gz
ERROR: unable to select packages:
  python-dev (no such package):
    required by: world[python-dev]
```

**Comment by jon-betts** [comment]
Ok, this all looks like it's working... which is a little odd, as I don't think `ws4py` should even compile. It might be that we skirt around the parts that aren't Python 3.7+ compatible, or that our tests don't exercise it.

I've run `make dev` locally and everything starts, but I've not actually tested that things work.

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -42,8 +74,12 @@ jobs:
           ${{ runner.os }}-tox-backend-
     - name: Run tox
       run: tox --parallel auto -e checkformatting,tests,coverage
+
   functests:
+    name: Funcional tests
```
Comment (jon-betts) [inline .github/workflows/continuous-integration.yml:79]:
```suggestion
    name: Functional tests
```

File: requirements/docs.txt [hunk 2]
```diff
@@ -2,15 +2,15 @@
 # This file is autogenerated by pip-compile
 # To update, run:
 #
-#    pip-compile requirements/docs.in
+#    pip-compile /home/jon/projects/h/requirements/docs.in
```
Comment (marcospri) [inline requirements/docs.txt:5]: Full path here?

**Review by marcospri — APPROVED** [review]

File: requirements/docs.txt (hunk 2 above)
Comment (jon-betts) [reply requirements/docs.txt:5]: Oh that's unfortunate... yeah this is going to be due to specifying the files absolutely I'd guess...

Comment (jon-betts) [reply requirements/docs.txt:5]: Wait, why only this one file!?

**Pull request merged**